src/ipc/ipc-common/IPCProxyModelProperty.h

```diff
@@ -139,16 +139,21 @@ class IPCProxyModelProperty : public facelift::ModelProperty<ModelDataType>
                     --last;
                 }
 
-                QList<ModelDataType> list;
-                m_proxy.ipc()->sendMethodCallWithReturn(requestMemberID, list, first, last);
+                std::pair<int, QList<ModelDataType>> listPair;
+                m_proxy.ipc()->sendMethodCallWithReturn(requestMemberID, listPair, first, last);
 
-                Q_ASSERT(list.size() == (last - first + 1));
+                auto &list = listPair.second;
+                first = listPair.first;// the server could change the number of returned items
+                if(list.empty()) {     // and even return an empty list
```

Comment:
"or even return an empty list"

add space between 'if' and '('
add space between ; and //